# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1236454902 18000
#      Sat Mar 07 14:41:42 2009 -0500
# Node ID 379297a149f017d4453728ad8a0ecb96d9b7d5da
# Parent  80f044c83de39aaae48314cc53dc255cd202c5bb
utils.cc (octave_vsnprintf): avoid uninitialized variable warning

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,10 +1,13 @@
 2009-03-07  John W. Eaton  <jwe@octave.org>
 
+	* utils.cc (octave_vsnprintf): Avoid uninitialized variable
+	warning from GCC.
+
 	* DLD-FUNCTIONS/qz.cc (Fqz): Avoid "maybe clobbered by vfork"
 	warning from GCC.
 
 	* version.h (OCTAVE_VERSION): Now 3.1.54.
 	(OCTAVE_API_VERSION): Now api-v36.
 	(OCTAVE_RELEASE_DATE): Now 2009-03-07.
 
 	* octave.cc (verbose_usage): Include --doc-cache-file in option list.
diff --git a/src/utils.cc b/src/utils.cc
--- a/src/utils.cc
+++ b/src/utils.cc
@@ -1111,19 +1111,19 @@ octave_vsnprintf (const char *fmt, va_li
 {
   static const size_t initial_size = 100;
 
   static size_t size = initial_size;
 
   static char *buf = 0;
 
 #if defined (HAVE_C99_VSNPRINTF)
-  size_t nchars;
+  size_t nchars = 0;
 #else
-  int nchars;
+  int nchars = 0;
 #endif
 
   if (! buf)
     buf = new char [size];
 
   if (! buf)
     return 0;
 
